cpp/src/arrow/adapters/orc/adapter.h

```diff
@@ -149,7 +156,7 @@ class ARROW_EXPORT ORCFileWriter {
   /// \brief Creates a new ORC writer.
   ///
   /// \param[in] output_stream a pointer to the io::OutputStream to write into
-  /// \return the returned writer object
+  /// \return the returned writer object or Status if error occurs
```

Comment:
There is no need to state this explicitly, as `Result<>` can implicitly convey an error.